### DISCUSSION TITLE
tutorials/3d/3d_text

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#183] [User-contributed notes]
# tutorials/3d/3d_text

Introduction: In a project, there may be times when text needs to be created as part of a 3D scene and not just in the HUD. Godot provides 2 methods to do this: the Label3D node and the TextMesh re...

https://docs.godotengine.org/en/stable/tutorials/3d/3d_text.html

**Comment by MajorMcDoom** [comment]
There is one more approach which should be mentioned. It's a little niche, but nevertheless valid:

Assigning a material with a `ViewportTexture` that shows the contents of a `SubViewport` with 2D controls.

**Advantages:**
- You can show it on any arbitrary 3D geometry that doesn't have to be flat.
- You can leverage all the features of `RichTextLabel`.
- You can show any kind of 2D control, not just text.
- You can reuse existing 2D UI resources in a 3D context.

**Disadvantages:**
- Complex to setup.
- Not as performant.
